pkg/resource/rtestutils/assertions.go

```diff
@@ -289,6 +289,95 @@ func AssertLength[R ResourceWithRD](ctx context.Context, t *testing.T, st state.
 	}
 }
 
+// AssertOptionalResources asserts that at least one of the resources exists and passes the assertionFunc.
+func AssertOptionalResources[R ResourceWithRD](
```

Comment:
nit: better naming?

```suggestion
func AssertOneOfResources[R ResourceWithRD](
```


also I wonder if we could make it simply an option to `AssertResources` with "how many 'passed' is enough?"